returnn/datasets/basic.py

```diff
@@ -170,9 +175,9 @@ def __init__(
         self._chunking = chunking
         self.chunk_size, self.chunk_step, self.custom_chunking_func = self._parse_chunking(chunking)
         self._context_window = context_window
-        assert 0 <= _shard_index < _num_shards
-        self._num_shards = _num_shards
-        self._shard_index = _shard_index
+        assert 0 <= shard_index < num_shards
+        self.num_shards = num_shards
+        self.shard_index = shard_index
```

Comment:
```suggestion
        self.set_shard_idx_and_num_shards(shard_index, num_shards)
```
Slightly cleaner and reuses code, but it's also fine as is. Your choice.

Reply:
That function has additional asserts that rely on correct initialization. Not for now.

Reply:
> That function has additional asserts that rely on correct initialization. Not for now.

I don't understand the comment. What correct initialization? What do you mean by "not for now"?